### PR TITLE
Dev/1.0

### PR DESCRIPTION
**Pull request by alexjhawk**

### CODE REVIEW

**Comment by alexjhawk** [comment]
An implementation of each of the three abstract classes can be found in my test class at https://github.com/hms-networks/sc-flexy-retry-lib/blob/test/1.0/src/test/java/com/hms_networks/americas/sc/retry/test/AutomaticRetryTestMain.java

File: README.md [hunk 1]
```diff
@@ -1,2 +1,23 @@
...
+## [Description](#table-of-contents)
+
+A library for managing the automatic retry of code using a linear, exponential or other assigned backoff algorithm.
+
+## [Developer Documentation](#table-of-contents)
+
+Developer documentation is available in Javadoc format found in the release package https://github.com/hms-networks/sc-flexy-retry-lib/releases.
+
+## [Dependencies](#table-of-contents)
+None
```
Comment (TomKimsey) [inline README.md]:
```suggestion

None
```

Comment (TomKimsey) [reply README.md]: For consistency

**Review by TomKimsey — CHANGES_REQUESTED** [review]
I would expect some users would want a constant retry algorithm (ie retry after 5 seconds every time it fails. 

Also you should probably add the github actions to this pr so that code style is verified.

Why was the decision made to implement abstract classes which a user has to implement rather than just passing a user written method as a callable or some other 1.4 friendly way? In doing that the user doesnt need to implement anything on the library side.

File: README.md (hunk 1 above)
Comment (alexjhawk) [reply README.md]: fixed

**Comment by alexjhawk** [comment]
> I would expect some users would want a constant retry algorithm (ie retry after 5 seconds every time it fails.
> 
> Also you should probably add the github actions to this pr so that code style is verified.
> 
> Why was the decision made to implement abstract classes which a user has to implement rather than just passing a user written method as a callable or some other 1.4 friendly way? In doing that the user doesnt need to implement anything on the library side.

- Users can implement that using the standard AutomaticRetryCode.java class. They can specify the delay in the return value of getDelayMillisBeforeRetry(). Additionally, they would need to implement the code() method and the getMaxRetries() method. 
- GitHub actions have been added
- The decision was made to use abstract classes because it is a simple way to provide the setState() method which is integral in order to allow for proper retry or critical stop depending on what occurs. Additionally, the Runnable class in Java 1.4 is itself an abstract class and requires you to implement the method #run(). Our abstract class(es) are very similar, except that they add additional methods that determine the parameters of the retry algorithm.
   - EDIT: an example of setState() usage can be found at https://github.com/hms-networks/sc-flexy-retry-lib/blob/46937963d116a8519039275b756b19c1d016bf80/src/test/java/com/hms_networks/americas/sc/retry/test/AutomaticRetryTestMain.java#L54

File: src/main/java/com/hms_networks/americas/sc/retry/AutomaticRetryState.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  /** Public instance of {@link AutomaticRetryState} representing error (retry) state. */
+  public static final AutomaticRetryState ERROR_RETRY = new AutomaticRetryState(ERROR_RETRY_INT);
+
+  /** Public instance of {@link AutomaticRetryState} representing error (stop) state. */
+  public static final AutomaticRetryState ERROR_STOP = new AutomaticRetryState(ERROR_STOP_INT);
+
+  /** Public instance of {@link AutomaticRetryState} representing finished state. */
+  public static final AutomaticRetryState FINISHED = new AutomaticRetryState(FINISHED_INT);
+
+  /** Instance state integer */
```
Comment (oliver-walker-hms) [inline src/main/java/com/hms_networks/americas/sc/retry/AutomaticRetryState.java]: missing period

File: src/main/java/com/hms_networks/americas/sc/retry/AutomaticRetryResult.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      new AutomaticRetryResult(FAIL_RETRY_LIMIT_INT);
+
+  /**
+   * Public instance of {@link AutomaticRetryResult} representing a failing result, due to
+   * encountering a critical error.
+   */
+  public static final AutomaticRetryResult FAIL_CRITICAL_STOP =
+      new AutomaticRetryResult(FAIL_CRITICAL_STOP_INT);
+
+  /** Instance result integer */
```
Comment (oliver-walker-hms) [inline src/main/java/com/hms_networks/americas/sc/retry/AutomaticRetryResult.java]: Missing period

**Review by oliver-walker-hms — CHANGES_REQUESTED** [review]

File: src/main/java/com/hms_networks/americas/sc/retry/AutomaticRetryState.java (hunk 2 above)
Comment (alexjhawk) [reply src/main/java/com/hms_networks/americas/sc/retry/AutomaticRetryState.java]: fixed/added

File: src/main/java/com/hms_networks/americas/sc/retry/AutomaticRetryResult.java (hunk 3 above)
Comment (alexjhawk) [reply src/main/java/com/hms_networks/americas/sc/retry/AutomaticRetryResult.java]: fixed/added

**Review by oliver-walker-hms — DISMISSED** [review]

**Review by TomKimsey — DISMISSED** [review]

**Review by it-hms — DISMISSED** [review]
I approve. My only recommendation is to change the name of _code_ method to something more verbose/specific like _retry_action_ ( or something better). I cannot think of a better name right now. hopefully you can.

**Comment by alexjhawk** [comment]
> I approve. My only recommendation is to change the name of _code_ method to something more verbose/specific like _retry_action_ ( or something better). I cannot think of a better name right now. hopefully you can.

I've updated the method name to #codeToRetry()

**Review by it-hms — APPROVED** [review]

**Review by TomKimsey — APPROVED** [review]

**Review by oliver-walker-hms — APPROVED** [review]

**Pull request merged**